### PR TITLE
COMPUTE-413: dx toolkit support for dx-jobutil-get-identity-token

### PR DESCRIPTION
**Pull request by bayou-brogrammer**
JIRA = COMPUTE-413
---

Good portion of the changes are from running the generators. The 3 main files to focus on:

- src/python/dxpy/api.py
- src/python/test/test_dx_bash_helpers.py
- src/python/scripts/dx-jobutil-get-identity-token

### CODE REVIEW

**Review by saisreepamidi — APPROVED** [review]
At a high level it looks good to me, but I think its important to gets Kurts approval on this before you merge it, I lack domain knowledge around the CLI.

File: src/python/test/test_dx_bash_helpers.py
Comment (wormsik) [inline src/python/test/test_dx_bash_helpers.py:355]: I would consider creating a new test suite for tokens...I don't feel that this is the right place...what do you think @kpjensen ?

File: src/python/scripts/dx-jobutil-get-identity-token
Comment (wormsik) [inline src/python/scripts/dx-jobutil-get-identity-token:30]: can you please describe also supported formats of providint subject claims in help? it seems that you can either use this arg multiple times or put multiple claims and separate using comma...

File: src/python/dxpy/toolkit_version.py
Comment (wormsik) [inline src/python/dxpy/toolkit_version.py:1]: can you please revert version? I think we no longer use automatic bump...am I right @kpjensen ?

File: src/python/test/file_load/job_identity_token/README.md
Comment (wormsik) [inline src/python/test/file_load/job_identity_token/README.md:1]: can you please justify why is this applet saved in `file_load` directory?

File: src/python/test/file_load/job_identity_token/dxapp.json
Comment (wormsik) [inline src/python/test/file_load/job_identity_token/dxapp.json:13]: I would consider removing optional as if not provided, you'd get `subject_claims = ['']` which is not valid request

File: src/python/scripts/dx-jobutil-get-identity-token
Comment (wormsik) [inline src/python/scripts/dx-jobutil-get-identity-token:44]: shouldn't be the request field name `subject_claims`?

File: src/python/test/test_dx_bash_helpers.py
Comment (wormsik) [inline src/python/test/test_dx_bash_helpers.py:366]: not sure if we have to use patched bash helpers here...perhaps just plain `dx build` would do the trick? (so it is not dependent on this directory)

**Review by wormsik — CHANGES_REQUESTED** [review]

File: src/python/scripts/dx-jobutil-get-identity-token
Comment (bayou-brogrammer) [reply src/python/scripts/dx-jobutil-get-identity-token:44]: on the javascript side everything is `camelCase` but python is `snake case`. We decided as a team to have python keep python standards and the api adhere to js standards

File: src/python/test/test_dx_bash_helpers.py
Comment (bayou-brogrammer) [reply src/python/test/test_dx_bash_helpers.py:366]: this is where I was told to place the test. Its an external bash script to dx toolkit so i think having a test in bash_helpers makes sense

File: src/python/test/test_dx_bash_helpers.py
Comment (kpjensen) [reply src/python/test/test_dx_bash_helpers.py:357]: I think it's okay to leave here as similar python standalone test scripts for `dx-upload-all-outputs` exist here.

File: src/python/test/test_dx_bash_helpers.py
Comment (kpjensen) [reply src/python/test/test_dx_bash_helpers.py]: All of the standalone scripts are referred to as "bash helpers", even though this one is pure python. This helper ensures the remote applet installs `dxpy` to be the same as the checkout running the tests.

File: src/python/test/jwt/README.md
Comment (kpjensen) [inline src/python/test/jwt/README.md:1]:
```suggestion
This is a testing app for the dx-jobutil-get-identity-token script. 
```

**Review by wormsik — APPROVED** [review]

**Comment by wormsik** [comment]
@bayou-brogrammer Thanks for addressing my comments! I have nothing left on my side and just let final approval on @kpjensen .

**Pull request merged**
